sdk/openfeature-nodejs-provider/src/DevCycleProvider.ts

```diff
@@ -0,0 +1,277 @@
+    ): Promise<ResolutionDetails<O>> {
+        const dvcVariable = this.dvcClient.variable(
+            this.dvcUserFromContext(context),
+            flagKey,
+            defaultValue
+        )
+        return this.resultFromDVCVariable<O>(
+            dvcVariable instanceof Promise
+                ? await dvcVariable
+                : dvcVariable
+        )
+    }
+
+    /**
+     * Resolve a boolean OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveBooleanEvaluation(
+        flagKey: string,
+        defaultValue: boolean,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<boolean>> {
+        return this.getDVCVariable(flagKey, defaultValue, context)
+    }
+
+    /**
+     * Resolve a string OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveStringEvaluation(
+        flagKey: string,
+        defaultValue: string,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<string>> {
+        return this.getDVCVariable(flagKey, defaultValue, context)
+    }
+
+    /**
+     * Resolve a number OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveNumberEvaluation(
+        flagKey: string,
+        defaultValue: number,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<number>> {
+        return this.getDVCVariable(flagKey, defaultValue, context)
+    }
+
+    /**
+     * Resolve a object OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveObjectEvaluation<T extends JsonValue>(
+        flagKey: string,
+        defaultValue: T,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<T>> {
+        return this.getDVCVariable(flagKey, this.defaultValueFromJsonValue(defaultValue), context)
+    }
+
+    /**
+     * Convert a OpenFeature JsonValue default value into DVCJSON default value for evaluation.
+     * @param jsonValue
+     * @private
+     */
+    private defaultValueFromJsonValue(jsonValue: JsonValue): DVCJSON {
+        if (typeof jsonValue !== 'object' || Array.isArray(jsonValue)) {
+            throw new Error('DevCycle only supports object values for JSON flags')
+        }
+        if (!jsonValue) {
+            throw new Error('DevCycle does not support null default values for JSON flags')
+        }
+
+        // Hard casting here because our DVCJSON typing enforces a flat object when we actually support
+        // a JSON Object of any depth. Will be fixed soon.
+        return jsonValue as DVCJSON
+    }
+
+    /**
+     * Convert a DVCVariable result into a OpenFeature ResolutionDetails.
+     * TODO: add support for variant / reason / and more error codes from DVC.
```

Comment:
Do you have a ticket to handle this TODO or should we fill this in now?

Reply:
I'll setup a ticket, this will require updating the NodeSDK to throw these errors instead of swallowing them and returning the default.